aten/src/ATen/core/type.cpp

```diff
@@ -600,13 +616,19 @@ bool TensorType::matchTensor(const at::Tensor& t) {
   }
   // Here we know t.defined() == true and compare all other properties.
   bool rg = at::GradMode::is_enabled() && t.requires_grad();
-  bool matched_strides = (!t.has_storage() && !stride_properties().isComplete())
-    || stride_properties() == computeStrideProps(t.sizes(), t.strides(), t.is_contiguous());
-  return scalarType().value_or(t.scalar_type()) == t.scalar_type()
-    && device().value_or(t.device()) == t.device()
-    && requiresGrad().value_or(rg) == rg
-    && matched_strides
-    && is_null_or_equal(sizes().concrete_sizes(), t.sizes());
+
+  if (requiresGrad().value_or(rg) != rg
+    || device().value_or(t.device()) != t.device()
+    || scalarType().value_or(t.scalar_type()) != t.scalar_type()
+    || !is_null_or_equal(sizes().concrete_sizes(), t.sizes())) {
+      return false;
+    }
+
+    if (!t.has_storage() && !stride_properties().isComplete()) {
+      return true;
+    }
+
+    return t.is_contiguous() ? matchStrides(this, t) : computeStrideProps(t.sizes(), false) == stride_properties();
```

Comment:
I'm confused by `computeStrideProps(t.sizes(), false)`.  How is that equivalent to the previous code?

Reply:
`computeStrideProps` also has a fast path that depends on tensor's contiguity and the `t.is_contiguous()` check makes sure we only go to `computeStrideProps` if we have a non-contiguous tensor. Maybe it makes sense to rip out the fast path from `computeStrideProps` since now it's handled by `matchStrides` ? @bertmaher  @eellison